### PR TITLE
[b4.4] ldmsd row cache cleanup

### PR DESCRIPTION
**Pull request by narategithub**

### CODE REVIEW

**Comment by narategithub** [comment]
This passed the associated test case: [ldmsd_decomp_static_rowcache_test](https://github.com/ovis-hpc/ldms-test/blob/b4.4/ldmsd_decomp_static_rowcache_test).

**Comment by nichamon** [comment]
Resolve #1483

File: ldms/src/ldmsd/ldmsd_row_cache.c [hunk 1]
```diff
@@ -247,13 +255,44 @@ void ldmsd_row_cache_idx_free(ldmsd_row_cache_idx_t idx)
...
+		LIST_REMOVE(g, bucket_entry);
+		rbt_del(&rcache->group_tree, &g->rbn);
+		#if 0
+		ldmsd_log(LDMSD_LINFO, "freeing group\n");
+		#endif
+		while ((rbn = rbt_min(&g->row_tree))) {
+			rbt_del(&g->row_tree, rbn);
+			cent = container_of(rbn, struct ldmsd_row_cache_entry_s, rbn);
+			ldmsd_row_cache_idx_free(cent->idx);
+			#if 0
```
Comment (tom95858) [inline ldms/src/ldmsd/ldmsd_row_cache.c:274]: We should probably get rid of this debug code.

File: ldms/src/ldmsd/ldmsd_row_cache.c [hunk 2]
```diff
@@ -155,6 +156,13 @@ ldmsd_row_cache_t ldmsd_row_cache_create(ldmsd_strgp_t strgp, int row_limit)
 	rcache->row_limit = row_limit;
 	rbt_init(&rcache->group_tree, tree_comparator);
 	pthread_mutex_init(&rcache->lock, NULL);
+	LIST_INIT(&rcache->group_bucket[0]);
```
Comment (tom95858) [inline ldms/src/ldmsd/ldmsd_row_cache.c:159]: We could use some comment on why there are three bucket groups.

File: ldms/src/ldmsd/ldmsd_row_cache.c [hunk 3]
```diff
@@ -262,6 +301,41 @@ int ldmsd_row_cache(ldmsd_row_cache_t rcache,
...
+	if (rc) {
+		rc = errno;
+		goto out;
+	}
+
+	if (rcache->cfg_timeout.tv_sec == 0 && rcache->cfg_timeout.tv_nsec == 0)
+		goto skip_cleanup;
+
+	/* process buckets */
+	count = 3;
```
Comment (tom95858) [inline ldms/src/ldmsd/ldmsd_row_cache.c:314]: This should be sizeof(rcache->group_bucket). We have this _magic_ constant 3 running around this logic.

**Pull request merged**